drivers/ethernet/eth_stm32_hal.c

```diff
@@ -394,6 +403,8 @@ static struct eth_stm32_hal_dev_data eth0_data = {
 			.RxMode = ETH_RXINTERRUPT_MODE,
 			.ChecksumMode = ETH_CHECKSUM_BY_SOFTWARE,
 			.MediaInterface = ETH_MEDIA_INTERFACE_RMII,
+			.DuplexMode = ETH_MODE_FULLDUPLEX,
+			.Speed = ETH_SPEED_100M,
```

Comment:
Will this disable auto negotiation of link speed and duplex mode?  Although we're in 2017, I can imagine this being plugged to a crusty switch that will only negotiate a half-duplex 10M link.

Reply:
You're right, I missed these are defined after auto negotiation. I'll remove this. Thanks!